File: Python3/938.py
# https://leetcode.com/problems/range-sum-of-bst
class Solution:
    def rangeSumBST(self, root: TreeNode, L: int, R: int) -> int:
        sum = 0
        
        stack = [root]
        
        while stack:
            
            node = stack.pop()
        
            if L <= node.val <= R:
                sum += node.val

            if node.val > L and node.left is not None:
                stack.append(node.left)
            if node.val < R and node.right is not None:
                stack.append(node.right)
        
        return sum

File: Go/938.go
/**
https://leetcode.com/problems/range-sum-of-bst
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func rangeSumBST(root *TreeNode, L int, R int) int {
    
    sum := 0
    
    if root.Val >= L && root.Val <= R {
        sum += root.Val
    }
    
    if root.Val > L && root.Left != nil {
        sum += rangeSumBST(root.Left, L, R)
    }
    
    if root.Val < R && root.Right != nil {
        sum += rangeSumBST(root.Right, L, R)
    }
    
    return sum
    
}
